pkg/util/vsphere/vm/cdrom.go

```diff
@@ -43,7 +43,7 @@ func UpdateCdromDeviceChanges(
 	for _, specCdrom := range vmCtx.VM.Spec.Cdrom {
 		imageRef := specCdrom.Image
 		// Sync the content library file if needed to connect the CD-ROM device.
-		syncFile := specCdrom.Connected
+		syncFile := *specCdrom.Connected
```

Comment:
With these fields changed to `*bool`, is a nil check needed here? With something like:
```go
func boolVal(v *bool) bool {
	if v == nil {
		return false
	}
	return *v
}
...
syncFile := boolVal(specCdrom.Connected)
```

Reply:
With the API defaulting this field to true, I don't think it could be nil? I also tried applying a VM YAML config with `connected: null` and found the field stored as the default true value in etcd. 

I agree it's always safe to double-check, and I'll use `pkg/util/ptr` to get its value safely.

Reply:
Updated with `ptr.Deref(specCdrom.Connected)`.